### PR TITLE
allow headless mode

### PR DESCRIPTION
**Pull request by rschmied**
- moved `--login-only` to `--headless`
- print auth response data as JSON via command line switch
- make certain error scenarios more robust
- fix unit test
- allow to specify AnyConnect version on CLI
- updated dependency versions (`pyproject.toml`)
- linted `README.md`
- added Makefile and some version handling

### CODE REVIEW

File: openconnect_sso/authenticator.py [hunk 1]
```diff
@@ -230,3 +244,6 @@ def _create_auth_finish_request(host, auth_info, sso_token):
 
 class AuthenticationCompleted(AuthenticationState):
     pass
+
+class NoSSOEndpointError(Exception):
```
Comment (rschmied) [inline openconnect_sso/authenticator.py]: I guess this could also be named `AuthResponseError` or similar

**Comment by rschmied** [comment]
Open to suggestions regarding naming and some cleanup, as required (e.g. sorting of imports or the `VERSION` var in the Makefile). Also, the versioning in `version.py` could be considered somewhat experimental. I was just trying to get a better version without messing with the upstream version.

**Comment by vlaci** [comment]
Thanks for your work!
I may not have time to look into it more deeply before the weekend.

File: openconnect_sso/authenticator.py [hunk 2]
```diff
@@ -173,6 +181,9 @@ class AuthCompleteResponse:
     session_token = attr.ib(converter=str)
     server_cert_hash = attr.ib(converter=str)
 
+    def asdict(self):
+        return self.__dict__
```
Comment (rschmied) [inline openconnect_sso/authenticator.py]: I think I should replace this with `attr.asdict(self)`
https://www.attrs.org/en/stable/api.html#attr.asdict

**Comment by vlaci** [comment]
I have integrated this feature however renamed it to `--authenticate` and made the json output opt-in (see changelog of release).

I have also implemented a Makefile inspired by this PR. Sorry for taking so long.

**Pull request closed without merging**